### PR TITLE
Absorb types into `@vocab/core`

### PR DESCRIPTION
**Pull request by mrm007**
`@vocab/types` re-exports all types from `@vocab/core` and will be deprecated and removed in the near future.

Once this is merged, #138 will automatically change its target to master.
Once #138 is merged, #139 will automatically change its target to master.

### CODE REVIEW

File: packages/core/src/config.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { UserConfig } from '@vocab/types';
+import type { UserConfig } from './types';
```
Comment (askoufis) [inline packages/core/src/config.ts:1]: Worth turning on the ESLint rule for this IMO (in a separate PR).

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -11,4 +11,4 @@ export {
   loadAllTranslations,
   loadTranslation,
 } from './load-translations';
-export type { TranslationFile } from '@vocab/types';
+export * from './types';
```
Comment (askoufis) [inline packages/core/src/index.ts:14]: Should be an `export type` with all the types listed explicitly, or does it not matter in this case?

File: packages/core/src/compile.ts [hunk 3]
```diff
@@ -150,9 +149,10 @@ function serialiseTranslationRuntime(
 
     if (Object.keys(params).length > 0) {
       const formatGeneric = hasTags ? '<T = string>' : '';
-      const formatReturn = hasTags
-        ? 'string | T | Array<string | T>'
-        : 'string';
```
Comment (askoufis) [inline packages/core/src/compile.ts:155]: I know it's a small change, but can we move this to a separate PR (and merge it before releasing this PR)? IMO moving types to the core package feels big enough that it should be a standalone change.

File: .changeset/new-starfishes-glow.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+'@vocab/types': patch
+'@vocab/core': patch
+---
+
+Remove [unused] `ICUFormatResult` type
+
+[unused]: https://github.com/search?q=+%2FICUFormatResult%2F&type=code
```
Comment (askoufis) [inline .changeset/new-starfishes-glow.md:8]: Same on this, could it be moved to a separate PR that just removes it from `@vocab/types`?

File: packages/core/src/index.ts (hunk 2 above)
Comment (mrm007) [reply packages/core/src/index.ts:14]: `export type` was [introduced in TypeScript 5.0](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#support-for-export-type). It would fail for projects on <5.0. In fact, that's why I made this commit [`44806cd` (#137)](https://github.com/seek-oss/vocab/pull/137/commits/44806cdc3012bc629557f58c1fe697b345c75801), because VS Code was fine, but `pnpm lint` failed because it was using v4.9.5.

File: .changeset/new-starfishes-glow.md (hunk 4 above)
Comment (mrm007) [reply .changeset/new-starfishes-glow.md:8]: Nobody uses that type. https://github.com/search?q=+%2FICUFormatResult%2F&type=code

File: packages/core/src/index.ts (hunk 2 above)
Comment (askoufis) [reply packages/core/src/index.ts:14]: Oh, I just meant `export type { /* all the types here */ } from './types'`. Sucks that `export type *` was only supported very recently.

File: .changeset/new-starfishes-glow.md (hunk 4 above)
Comment (askoufis) [reply .changeset/new-starfishes-glow.md:8]: Sure, but I still feel like it doesn't belong in this PR.

File: packages/core/src/index.ts (hunk 2 above)
Comment (mrm007) [reply packages/core/src/index.ts:14]: In that case, it doesn't matter. In the output `.d.ts` file `export X` and `export type X`  work the same way.

File: packages/core/src/config.ts (hunk 1 above)
Comment (mrm007) [reply packages/core/src/config.ts:1]: https://github.com/seek-oss/eslint-config-seek/pull/96

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (askoufis) [inline .vscode/settings.json:2]: Does this need to be configured per-repo? Seems like a sensible default for a user config. Would prefer to minimize editor-specific config, but I use NeoVim so I don't really have a voice here.

Comment (mrm007) [reply .vscode/settings.json:2]: Yes, or every user has to have it in their user settings.

**Review by askoufis — APPROVED** [review]
